### PR TITLE
TODO App with context-api

### PR DESCRIPTION
**Pull request by haroldmud**

### CODE REVIEW

File: src/components/List.jsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import React from 'react'
+import { useContext } from 'react'
```
Comment (IbrahimBagalwa) [inline src/components/List.jsx:2]: You no longer need to import React from "react". Starting from the release 17 of React, JSX is automatically transformed without using React.createElement.
However, other exports like hooks must be imported.

File: src/feature/listContext.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        setCheck(idx)
+        collection.checked = true
+      }else{setCheck(null)
+        collection.checked = false
+      } 
+  }
+
+  function handleNewItem(x){
+    setNewItem(x.target.value)
+  }
```
Comment (IbrahimBagalwa) [inline src/feature/listContext.js:54]: here as well

File: src/feature/listContext.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+function handleDiverse(){
+  setEdit(null); setCheck(null); setNewItem("");
+}
+
+const value = {list,item,check,edit,alerte,limit,newItem, handleList, handleCheck, handleDeletion, handleEdit, handleItem, handleNewItem, handleDiverse}
+    return(
+    <ListContext.Provider value={value}>
+      {props.children}
```
[the remark this answers is not in the document]
Comment (haroldmud) [reply src/feature/listContext.js:70]: @IbrahimBagalwa  this is the best practice actually

File: src/components/List.jsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+              </ul>
+            </div>
+          </div>
+        </div>
+      </div>
+   </section>
+  )
+}
+
+export default List
```
[the remark this answers is not in the document]
Comment (haroldmud) [reply src/components/List.jsx:65]: @IbrahimBagalwa it's less than one hundred line, and this is a custom component.

File: src/index.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
```
[the remark this answers is not in the document]
Comment (haroldmud) [reply src/index.js:1]: image

**Review by IbrahimBagalwa — APPROVED** [review]

**Pull request merged**